core/trino-spi/src/main/java/io/trino/spi/connector/ConnectorMetadata.java

```diff
@@ -545,6 +559,50 @@ default void finishUpdate(ConnectorSession session, ConnectorTableHandle tableHa
         throw new TrinoException(NOT_SUPPORTED, "This connector does not support updates");
     }
 
+    /**
+     * Return the row update paradigm supported by the connector on the table, or
+     * {@link RowChangeParadigm#ROW_CHANGE_NOT_SUPPORTED} if row change is not supported.
+     * @param session The session in which to start the update operation.
+     * @param tableHandle A ConnectorTableHandle for the table to be updated.
+     * @return Return the supported row update paradigm.
+     */
+    default RowChangeParadigm getRowChangeParadigm(ConnectorSession session, ConnectorTableHandle tableHandle)
+    {
+        return ROW_CHANGE_NOT_SUPPORTED;
+    }
+
+    /**
+     * Get the column handle that will generate row IDs for the merge operation.
+     * These IDs will be passed to the {@code storeMergedRows()} method of the
+     * {@link io.trino.spi.connector.ConnectorMergeSink} that created them.
+     */
+    default ColumnHandle getMergeRowIdColumnHandle(ConnectorSession session, ConnectorTableHandle tableHandle, MergeDetails mergeDetails)
+    {
+        throw new TrinoException(NOT_SUPPORTED, "This connector does not support merges");
+    }
+
+    /**
+     * Do whatever is necessary to start an MERGE query, returning the {@link ConnectorMergeTableHandle}
+     * instance that will be passed to the PageSink, and to the {@link #finishMerge} method.
+     * @return Returns a ConnectorTableHandle that will be passed to split generation, and to the
+     * {@link #finishMerge} method.
+     */
+    default ConnectorMergeTableHandle beginMerge(ConnectorSession session, ConnectorTableHandle tableHandle, MergeDetails mergeDetails)
+    {
+        throw new TrinoException(NOT_SUPPORTED, "This connector does not support merges");
+    }
+
+    /**
+     * Finish an update query
+     * @param tableHandle
+     * @param fragments all fragments returned by {@link UpdatablePageSource#finish()}
+     * @param computedStatistics statistics for the table and partitions
```

Comment:
what partitions?  (partition is not a SPI concept)

will this be provided for all partitions we write to?
like for insert case?

Reply:
> what partitions? (partition is not a SPI concept)

Good point - - only the connector understands how to interpret the computedStatistics.  Changed to "statistics for the table, meaningful only to the connector that produced them."

> will this be provided for all partitions we write to?
> like for insert case?

Yes.  But we mustn't mention partitions 😆